scripts/core/context.yml

```diff
@@ -65,19 +65,16 @@ class: $xContext
 name: $x_context_info_t
 etors:
     - name: NUM_DEVICES
-      value: "1"
       desc: "[uint32_t] The number of the devices in the context"
     - name: DEVICES
-      value: "2"
       desc: "[$x_context_handle_t...] The array of the device handles in the context"
+    - name: REFERENCE_COUNT
```

Comment:
Is this useful in scenarios besides simple tests? This sort of functionality can suffer from time of check/time of use (TOCTOU) races.

For testing, instead of just checking if `fetch_and_add` worked, it would be more useful to test the outcome - whether a proper sequence of calls appropriately retains or deallocates memory. Typically I would do this by integrating valgrind, but in UR case we'd probably need an API to query heap statistics.

Reply:
I'm inclined to agree, these queries exist in OpenCL with the caveat that `The reference count returned should be considered immediately stale. It is unsuitable for general use in applications. This feature is provided for identifying memory leaks. `.

Reply:
My experience is that people rarely read docs and are happy to misuse an API if it appears to help with their problem at first glance. In this case, I'm not even sure if you could use this for leak checking. I can't imagine it would be safe to call this on an object that's already released. You'd be reading from freed/unitialized memory, or, even worse, from another allocation.

@kswiecicki has been working on a validation layer to track object lifetimes, #289, which I think is going to be a better solution for this particular problem.

Reply:
Through some discussions we think that it's useful to assert that the reference count has changed after a release/retain entry point. During testing we are eliminating the concern for a multi-threaded context and thus race conditions like you suggested. I have however updated the documentation to clarify the danger with these queries.

Reply:
Well, I'm going to choose my battles, but I still don't entirely agree that this functionality is useful for detecting leaks :)